### ISSUE TITLE
Something wrong with catching errors from realtime subscriptions

### ISSUE THREAD

**Issue by geoguide** [supabase/realtime-js#416]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When a table doesn't exist or realtime isn't enabled you cannot get errors without having typescript errors. 
## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

From help on the discord I found I can do this to get the errors:

```
.on(
  ..
)
  .subscribe((status, error) => console.info(status, error))
  .on('system', { event: 'error' }, (error: any) => {
      console.info('system', JSON.stringify(error))
})
````

but this doesn't appear to be documented and throws this typescript error:
<img width="833" alt="image" src="https://github.com/user-attachments/assets/94859e0e-d210-4ade-90d2-27651bb4ede7">

## Expected behavior

At least allow the typescript to accept the 'system' in this case so that I can get the errors in my code, and at best show how to handle this in the docs

## System information

- OS: macOS
- Browser Any
- Version of supabase-js: 2.40
- Version of Node.js: 20.4

**Comment by filipecabaco** [comment]
Will close since it's available in version 2.10.5 - https://github.com/supabase/realtime-js/issues/121

**Comment by geoguide** [comment]
Thanks @filipecabaco !!!
